src/main.cpp

```diff
@@ -872,6 +875,18 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
                          REJECT_INVALID, "bad-txns-version-too-low");
     }
 
+    // Check Overwinter flag has been set if transaction version 3 or later
+    if (tx.nVersion >= 3 && !tx.fOverwinter) {
+        return state.DoS(100, error("CheckTransaction(): version requires overwinter flag to be set"),
+                         REJECT_INVALID, "overwinter-flag-not-set");
```

Comment:
I'm not sure whether this is necessary; this should be a parser error. It probably doesn't *hurt*, though that's not really something I want to say about consensus rules...

Reply:
Hmm, it occurs to me that we should use the masked version in the signature hash, which means this boolean will get ORed in, which means we probably do want some kind of check (to defend against internal mistakes, not bad transactions from peers - those should be caught as parser errors).

I would probably implement the above in some `GetHeader()` function, rather than doing it inside the sighash code, because then we know it's being masked the same way everywhere.